src/carma-streets/parse_kafka_logs.py

```diff
@@ -124,7 +142,77 @@ def parse_timesync_to_csv(inputfile: Path, outputfile: Path):
-   
+
+def parse_map_to_csv(inputfile: Path, outputfile: Path):
+    """Function to parse MAP Kafka Topic log file and generate csv data of all time sync messages
+
+    Args:
+        inputfile (Path): Path to Kafka Topic log file
+        outputfile (Path): File name (excluding file extension) of desired csv file
+    """
+
+    map_msgs = parse_kafka_logs_as_type(inputfile, KafkaLogMessageType.MAP)
+    if  outputfile.exists():
+        print(f'Output file {outputfile} already exists. Overwritting file.')
+    #write data of interest to csv which will be used to produce plots
+    with open(outputfile, 'w', newline='') as write_obj:
+        csv_writer = writer(write_obj)
+        csv_writer.writerow(['Created Time(ms)', 'Timestamp(ms)', 'Map Data'])
+        skipped_messages = 0
+        #extract relevant elements from the json
+        for msg in map_msgs:
+            try:
+                csv_writer.writerow([msg.created_time, msg.json_message['metadata']['timestamp'], msg.json_message['map_data']])
+            except Exception as e:
+                print(f'Error {e} occurred while writing csv entry for kafka message {msg.json_message}. Skipping message.')
+        if skipped_messages == 0 :
+            print('Finished writing all entries successfully')
+        else:
+            print(f'WARNING: Skipped {skipped_messages} due to errors. Please inspect logs')
+def parse_sdsm_to_csv(inputfile: Path, outputfile: Path):
+    """Function to parse SDSM Kafka Topic log file and generate csv data of all time sync messages
+
+    Args:
+        inputfile (Path): Path to Kafka Topic log file
+        outputfile (Path): File name (excluding file extension) of desired csv file
+    """
+
+    sdsm_msgs = parse_kafka_logs_as_type(inputfile, KafkaLogMessageType.SDSM)
+    if  outputfile.exists():
+        print(f'Output file {outputfile} already exists. Overwritting file.')
+    #write data of interest to csv which will be used to produce plots
+    with open(outputfile, 'w', newline='') as write_obj:
+        csv_writer = writer(write_obj)
+        csv_writer.writerow(['Created Time(ms)', 'Timestamp(ms)',
+             'Message Count', 'Source ID', 'Equipement Type', 'Reference Position Longitude', 
+             'Reference Position Latitude', 'Reference Position Elevation','Objects'])
+        skipped_messages = 0
+        #extract relevant elements from the json
+        for msg in sdsm_msgs:
+            try:
+                epoch_time = datetime.datetime( msg.json_message['sdsm_time_stamp']['year'], \
+                    msg.json_message['sdsm_time_stamp']['month'], \
+                    msg.json_message['sdsm_time_stamp']['day'], \
+                    msg.json_message['sdsm_time_stamp']['hour'], \
+                    msg.json_message['sdsm_time_stamp']['minute'], \
+                    msg.json_message['sdsm_time_stamp']['second']//1000, \
+                    msg.json_message['sdsm_time_stamp']['second']%1000).timestamp()*1000
```

Comment:
Consider abstracting this away into a function to make the calling code more expressive:

```python
def get_sdsm_timestamp(json_data: dict):
    pass

timestamp = get_sdsm_timestamp(msg.json_message)
```